src/state_machine/coordinator/fire.rs

```diff
@@ -1680,12 +1690,47 @@ pub mod test {
             msg: Message::SignatureShareResponse(sig_share_response.clone()),
             sig: Default::default(),
         };
+
+        let response_info = coordinator
+            .message_nonces
+            .entry(coordinator.message.clone())
+            .or_default();
+        response_info.sign_wait_signer_ids.insert(0);
+
         coordinator
             .gather_sig_shares(&packet, signature_type)
             .unwrap();
         assert_eq!(1, coordinator.signature_shares.len());
 
-        // check that a duplicate private share is ignored
+        // check that a signature share is ignored due to not being on the wait list
```

Comment:
nit: as is it's not obvious that we are ignoring the share because it's missing from the wait list. We could swap those two: first try to process the package without it being in the wait list, check that `signature_shares` is empty, then add to the wait list and process the same message again, now expecting it to get into `signature_shares`

Reply:
Done